float32 x
float32 y
float32 speed_x
float32 speed_y